File: andr/dynamic-listview-picasa/src/com/google/auth/GoogleAuthenticator.java
package com.google.auth;

import org.apache.http.HttpResponse;
import org.apache.http.NameValuePair;
import org.apache.http.client.HttpClient;
import org.apache.http.client.entity.UrlEncodedFormEntity;
import org.apache.http.client.methods.HttpPost;
import org.apache.http.message.BasicNameValuePair;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.ArrayList;
import java.util.List;

/**
 * Created by callmewa on 8/10/13.
 */
public class GoogleAuthenticator {

    String authUrl = "https://www.google.com/accounts/ClientLogin";
    //String authParams = "accountType=HOSTED_OR_GOOGLE&service=lh2&source=gdata-PhotosPartialDemo&Email=platr.upload@gmail.com&Passwd=foodimages";

    //TODO: figure out a better way to host this
    static String authToken = null;

    public String authenticate(HttpClient client){
        HttpPost post = new HttpPost(authUrl);
        List<NameValuePair> nameValuePairs = new ArrayList<NameValuePair>(1);
        nameValuePairs.add(new BasicNameValuePair("Email", "platr.upload@gmail.com"));
        nameValuePairs.add(new BasicNameValuePair("Passwd", "foodimages"));
        nameValuePairs.add(new BasicNameValuePair("accountType", "HOSTED_OR_GOOGLE"));
        nameValuePairs.add(new BasicNameValuePair("source", "gdata-Demo"));
        nameValuePairs.add(new BasicNameValuePair("service", "lh2"));


        try {
            post.setEntity(new UrlEncodedFormEntity(nameValuePairs));
            HttpResponse response = client.execute(post);
            BufferedReader rd = new BufferedReader(new InputStreamReader(response.getEntity().getContent()));

            String line = "";
            while ((line = rd.readLine()) != null) {
                //System.out.println(line);
                if (line.startsWith("Auth=")) {
                    String authToken = line.substring(5);
                    GoogleAuthenticator.authToken = authToken;
                    return authToken;
                }
            }
        }
        catch (IOException e) {
            post.abort();
        } catch (Exception e){
            e.printStackTrace();
        }
        return null;
    }
}

File: ios/platr/platr/GoogleAuthenticator.m
//
//  GoogleAuthenticator.m
//  platr
//
//  Created by wa on 10/12/13.
//  Copyright (c) 2013 com.platr. All rights reserved.
//

#import "GoogleAuthenticator.h"
#import "LoadFeedData.h"
static NSString *AuthToken = nil;


@implementation GoogleAuthenticator

NSString *const authUrl = @"https://www.google.com/accounts/ClientLogin";
NSString *const authParams = @"accountType=HOSTED_OR_GOOGLE&service=lh2&source=gdata-Demo&Email=platr.upload@gmail.com&Passwd=foodimages";

+ (NSString *) getAuthToken
{
    return AuthToken;
}

- (NSString *) authenticate
{
    dispatch_queue_t requestQueue = dispatch_queue_create("RequestQueue", NULL);
    dispatch_async(requestQueue, ^{
        // Perform long running process
        NSData *postData = [authParams dataUsingEncoding: NSUTF8StringEncoding];
        NSMutableURLRequest *post = [[NSMutableURLRequest  alloc] initWithURL:[NSURL URLWithString:authUrl]];
        [post setHTTPMethod:@"POST"];
        [post setHTTPBody:postData];
        [post setValue:@"application/x-www-form-urlencoded" forHTTPHeaderField:@"Content-type"];
        NSURLResponse *response;
        NSError *error;
        NSData *responseData = [NSURLConnection sendSynchronousRequest:post returningResponse:&response error: &error ];
        NSString* responseStr = [[NSString alloc] initWithData:responseData
                                                      encoding:NSUTF8StringEncoding];
        
        NSLog(@"responseData: %@", responseStr);
        //NSLog(@"responseData: %@", responseData);
        
        NSArray *lines = [responseStr componentsSeparatedByString:@"\n"];
        for(NSString *line in lines){
            if([line hasPrefix:@"Auth="]){
                NSString *authToken = [line substringFromIndex:5];
                AuthToken = authToken;
            }
        }

        dispatch_async(dispatch_get_main_queue(), ^{
            // Update the UI
            NSLog(@"AuthToken: %@", AuthToken);
            LoadFeedData *loadFeed = [[LoadFeedData alloc] init];
            [loadFeed loadFeed];
            
        });
    });
    return nil;
}

@end
